qsimcirq/qsim_circuit.py

```diff
@@ -92,14 +94,26 @@ def _cirq_gate_kind(gate: cirq.ops.Gate):
     return qsim.kPhasedISwapPowGate
   if isinstance(gate, cirq.ops.FSimGate):
     return qsim.kFSimGate
+  if isinstance(gate, cirq.ops.TwoQubitDiagonalGate):
+    return qsim.kTwoQubitDiagonalGate
+  if isinstance(gate, cirq.ops.ThreeQubitDiagonalGate):
+    return qsim.kThreeQubitDiagonalGate
+  if isinstance(gate, cirq.ops.CCZPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+      return qsim.kCCZ
+    return qsim.kCCZPowGate
+  if isinstance(gate, cirq.ops.CCXPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+      return qsim.kCCX
+    return qsim.kCCXPowGate
+  if isinstance(gate, cirq.ops.CSwapGate):
+    return qsim.kCSwapGate
   if isinstance(gate, cirq.ops.MatrixGate):
```

Comment:
[No action required] With some extra tinkering in the python code, I think we can convert any non-decomposable Cirq gates into their matrix forms, allowing qsim to execute _any_ Cirq circuit with gates of up to 6 qubits.

I can take a look at this if you aren't already working on it.

Reply:
I'm not working on it. It would be great if you take a look at this.